x-pack/plugins/data_visualizer/public/application/index_data_visualizer/hooks/esql/use_data_visualizer_esql_data.tsx

```diff
@@ -229,6 +229,24 @@ export const useESQLDataVisualizerData = (
           } as QueryDslQueryContainer;
         }
       }
+
+      // Ensure that we don't query frozen data
+      if (filter.bool && Array.isArray(filter.bool?.must_not)) {
+        filter.bool.must_not.push({
+          term: {
+            _tier: 'data_frozen',
+          },
+        });
+      } else {
+        filter.bool = {
+          must_not: [
+            {
+              term: { _tier: 'data_frozen' },
+            },
+          ],
+        };
+      }
```

Comment:
If a bool object already exists containing a `must` but not `must_not` this will hit the `else` clause where a new bool object is created, wiping out the exiting `must`.

I think a series of ifs might be safer. something like

```ts
if (filter.bool === undefined) {
  filter.bool = Object.create(null);
}

if (filter.bool.must_not === undefined) {
  filter.bool.must_not = [];
}

filter.bool.must_not.push({
  term: { _tier: 'data_frozen' },
  })
```

Reply:
Updated here [`83ff92c` (#200325)](https://github.com/elastic/kibana/pull/200325/commits/83ff92cf43dcc4760842d686b47fd3076b731ba3)